grails-app/domain/com/unifina/domain/community/CommunitySecret.groovy

```diff
@@ -0,0 +1,37 @@
+package com.unifina.domain.community
+
+import com.unifina.domain.marketplace.Product
+import com.unifina.utils.HexIdGenerator
+import grails.compiler.GrailsCompileStatic
+import groovy.transform.ToString
+
+@ToString
+class CommunitySecret {
+	String id
+	// name to display for users.
+	String name
+	// secret shared by the community that enables automatic join.
+	String secret
+	// communityAddress is an Ethereum address of the community.
+	String communityAddress
+
+    static constraints = {
+		communityAddress(validator: Product.isEthereumAddress)
+    }
+	static mapping = {
+		id generator: HexIdGenerator.name
```

Comment:
Should actually use `IdGenerator` instead of `HexIdGenerator`, if there's no reason to do otherwise